src/tamm/tensor_impl.hpp

```diff
@@ -1066,30 +1069,30 @@ class DenseTensorImpl: public TensorImpl<T> {
 
   /// @todo Should this be GA_Nodeid() or GA_Proup_nodeid(GA_Get_pgroup(ga_))
   T* access_local_buf() override {
-#if defined(USE_UPCXX)
-    throw std::runtime_error("upcxx: dense tensor - access_local_buf unsupported");
-#else
     EXPECTS(allocation_status_ != AllocationStatus::invalid);
     T* ptr;
+#if defined(USE_UPCXX)
+    ptr = reinterpret_cast<T*>(local_gptr_.local());
```

Comment:
@jcorbino-lbl is there a reason `local_gptr_` was declared as a `upcxx::global_ptr<uint8_t>` instead of `upcxx::global_ptr<T>`? The latter seems more natural would avoid these ugly casts.  IIUC `element_size == sizeof(T)`, or am I missing something?

Reply:
I don't see why declaring `local_gptr_` **AND** making the pertinent changes after that will break something (thinking about the alignment of all the possible component types). I agree with you, having casts everywhere doesn't look pretty. Thanks for bringing that up!